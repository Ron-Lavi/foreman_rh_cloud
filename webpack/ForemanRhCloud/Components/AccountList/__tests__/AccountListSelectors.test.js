import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';
import {
  selectAccountsList,
  selectAccounts,
  selectPollingProcessID,
  selectAutoUploadEnabled,
} from '../AccountListSelectors';
import {
  pollingProcessID,
  accounts,
  autoUploadEnabled,
} from '../AccountList.fixtures';

const state = {
  ForemanRhCloud: {
    accountsList: {
      accounts,
      pollingProcessID,
      autoUploadEnabled,
    },
  },
};

const fixtures = {
  'should return AccountsList': () => selectAccountsList(state),
  'should return AccountList accounts': () => selectAccounts(state),
  'should return AccountList pollingProcessID': () =>
    selectPollingProcessID(state),
  'should return AccountList autoUploadEnabled': () =>
    selectAutoUploadEnabled(state),
};

describe('AccountList selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
