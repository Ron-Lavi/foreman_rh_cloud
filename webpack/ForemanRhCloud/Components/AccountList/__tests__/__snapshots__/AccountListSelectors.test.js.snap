// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountList selectors should return AccountList accounts 1`] = `
Object {
  "Account1": Object {
    "generate_report_status": "running",
    "label": "test_org1",
    "upload_report_status": "running",
  },
  "Account2": Object {
    "generate_report_status": "failure",
    "label": "test_org2",
    "upload_report_status": "unknown",
  },
  "Account3": Object {
    "generate_report_status": "running",
    "label": "test_org3",
    "upload_report_status": "success",
  },
}
`;

exports[`AccountList selectors should return AccountList autoUploadEnabled 1`] = `true`;

exports[`AccountList selectors should return AccountList pollingProcessID 1`] = `0`;

exports[`AccountList selectors should return AccountsList 1`] = `
Object {
  "accounts": Object {
    "Account1": Object {
      "generate_report_status": "running",
      "label": "test_org1",
      "upload_report_status": "running",
    },
    "Account2": Object {
      "generate_report_status": "failure",
      "label": "test_org2",
      "upload_report_status": "unknown",
    },
    "Account3": Object {
      "generate_report_status": "running",
      "label": "test_org3",
      "upload_report_status": "success",
    },
  },
  "autoUploadEnabled": true,
  "pollingProcessID": 0,
}
`;
